src/Raven.Server/Rachis/Leader.cs

```diff
@@ -748,14 +749,22 @@ public void Dispose()
                     {
                         if (await waitAsync == false)
                         {
-                            if (rachisMergedCommand.Consumed.Raise())
+                            commandAborted = rachisMergedCommand.Consumed.Raise();
+                            if (commandAborted)
                                 throw new TimeoutException($"Waited for {timeout} but the command was not applied in this time.");
 
                             // if the command is already dequeued we must let it continue to keep its context valid.
                             await rachisMergedCommand.Tcs.Task;
                         }
                     }
                 }
+                catch (Exception) when (rachisMergedCommand.Consumed.Raise() == false && commandAborted == false && rachisMergedCommand.Tcs.Task.IsCompleted == false)
+                {
+                    // If the command has already been dequeued, we must allow it to continue to maintain its context validity.  
```

Comment:
we already have this comment in the code above, which makes me wonder whether we want to move the code
```
await rachisMergedCommand.Tcs.Task;
``` 
to be part of the `RachisMergedCommand` dispose.
This way we cover all cases, no?

Reply:
In the code above, I could just throw the exception and let the code here handle that, but I preferred to keep it simple.
I will go with your suggestion.
Just two things for the record: 
* That means that on a transient error, like OutOfMemory, we will still throw this error to the handler task even if this error passed away after the awaiting.
I guess it is not an issue since, like OutOfMemory, the system is in stress, and it makes sense to get an exception about that, and it has the advantage of not swallowing errors.
* Also, when we wait for that task, this request is "blocked" and can't "EmptyQueue."
This is also the issue with the code above, and I think it is not a big issue since we can rely on more requests to come, so one of them will "EmptyQueue."